tools/tu_collector/tu_collector/tu_collector.py

```diff
@@ -400,22 +419,50 @@ def main():
     if args.logfile:
         with open(args.logfile, encoding="utf-8", errors="ignore") as f:
             compilation_db = json.load(f)
+    else:
+        compilation_db = [{
+            'file': '',
+            'command': args.command,
+            'directory': os.getcwd()}]
 
-        if args.filter:
+    if args.zip:
+        if args.logfile and args.filter:
             compilation_db = [
                 action for action in compilation_db if fnmatch.fnmatch(
                     action['file'], args.filter)]
-    else:
-        if args.filter:
+        else:
             LOG.warning('In case of using build command the filter has no '
                         'effect.')
-        compilation_db = [{
-            'file': '',
-            'command': args.command,
-            'directory': os.getcwd()}]
 
-    zip_tu_files(args.zip, compilation_db)
-    LOG.info("Done.")
+        zip_tu_files(args.zip, compilation_db)
+        LOG.info("Done.")
+    else:
```

Comment:
I think we should place the content of this "else" branch to a standalone function `get_dependent_sources()` so this functionality can also be used by other tools as a library function.